### PR TITLE
Issue #123421 chore: added onbeforesave trigger

### PR DESCRIPTION
**Pull request by praneettekdi**

### CODE REVIEW

File: site/controllers/itemform.php [hunk 1]
```diff
@@ -307,6 +307,10 @@ public function save($key = null, $urlVar = null)
 		// Jform tweaking - get data for extra fields jform.
 		$extra_jform_data = array_diff_key($all_jform_data, $validData);
 
+		$dispatcher        = JDispatcher::getInstance();
+		JPluginHelper::importPlugin("system");
+		$onBeforeTjucmSave = $dispatcher->trigger('onBeforeTjucmSave');
```
Comment (ankush-maherwal) [inline site/controllers/itemform.php]: cant we use 'onContentBeforeSave' ?

Do we really need to add custom plugin trigger?

Comment (praneettekdi) [reply site/controllers/itemform.php]: We need custom trigger as it should trigger only before tjucm save

Comment (ankush-maherwal) [reply site/controllers/itemform.php]: we can use $context as the first param of this predefined content trigger

https://docs.joomla.org/Plugin/Events/Content#onContentBeforeSave

Comment (praneettekdi) [reply site/controllers/itemform.php]: I have passed what is needed for my use case, if you want I can put that param.

Comment (praneettekdi) [reply site/controllers/itemform.php]: @ankush-maherwal can we have the closure on this ?

**Comment by ankush-maherwal** [comment]
tjucmOnBeforeSaveItem added in the package

**Pull request closed without merging**